app/controllers/admin/creator_settings_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Admin
+  class CreatorSettingsController < Admin::ApplicationController
+    before_action :extra_authorization, only: [:create]
+
+    ALLOWED_PARAMS = %i[community_name logo_svg primary_brand_color_hex invite_only_mode public].freeze
+
+    def new; end
+
+    def create
+      ::Settings::Community.community_name = settings_params[:community_name]
+      ::Settings::General.logo_svg = settings_params[:logo_svg]
+      ::Settings::UserExperience.primary_brand_color_hex = settings_params[:primary_brand_color_hex]
+      ::Settings::Authentication.invite_only_mode = settings_params[:invite_only]
+      ::Settings::UserExperience.public = settings_params[:public]
+      current_user.update!(saw_onboarding: true)
+      redirect_to root_path
```

Comment:
This is updating the user as expected (I can see that `saw_onboarding` is true in my console and browser), but after completing the Creator Setup view, my user _still_ gets thrown into the traditional onboarding slides upon the redirect to the home page. This was a nonissue in the past, but started occurring within the past week. 🤔

Reply:
We've had onboarding loops for a while, though they don't seem to happen consistently, see https://github.com/forem/forem/issues/14483

Reply:
@juliannatetreault looking at the code, its probably because in the `onboardingRedirectCheck.js` file, the `[onboardingSkippable` check is not met](https://github.com/forem/forem/blob/b3e2670a27896e63bb4f7b8f9e34cc5a1122ee95/app/javascript/packs/onboardingRedirectCheck.jsx#L21-L27).

```
function onboardingSkippable(currentUser) {
  return (
    currentUser.saw_onboarding &&
    currentUser.checked_code_of_conduct &&
    currentUser.checked_terms_and_conditions
  );
}
``` 

`saw_onboarding` would be true but `checked_code_of_conduct` and `checked_terms_and_conditions` would still be false. Hence, the app would think that we still need to go through a user onboarding.

To do a quick check, set these to true and it shouldn't redirect back to the user onboarding any longer.

Now onto the question of how do we solve this?
It feels weird to show the creator a code of conduct and terms of conditions for their own Forem, when they will be the ones setting it, hence to me it makes logical sense to just set these to true in the controller. However, let's double check the implications of this with Peter/Armando please.

Reply:
Thank you both so much for taking a look! This all makes sense. Updating the current, creator user in the controller to having seen both the CoC and Terms and Conditions did the trick!

> However, let's double check the implications of this with Peter/Armando please.

Will do, @Ridhwana 👍